### ISSUE TITLE
Any updates on the low storage space fix?

### ISSUE THREAD

**Issue by xGary** [googlesamples/android-vision#45]
Any updates on the low storage space fix? It has been a while and it is annoying if I have to make users clear out up to 10% of their storage to enable a small feature.

**Comment by pm0733464** [comment]
Sorry about that.  We are working on this, but don't yet have a launch date to announce at this time.

**Comment by mitchross** [comment]
This is a workaround you can use. https://gist.github.com/mitchross/b012bd72a9cd3b1f92c7

**Comment by gunhansancar** [comment]
@pm0733464 we are included the vision barcode api in our new application however this very issue is really annoying. We don't want to use the other very well known open source library for barcode reading. First because it is not multi-orientation, second it is not an official api. But according to the vision api implementation we have to include it anyway because we should provide this feature for all the users. But as long as you do not change that low storage control we have to include both of the libraries because of the wrong desicions that you made at the beginning.

@mitchross Do you suggest that before trying to use the barcode reader first check if 10% of storage available? 

What kind of workaround is this?

For a nowadays standard device with 16gb internal memory, the library requires 1.6gb of free space for what exactly? How many megabytes does it need to downlaod? 1mb? 10mb?

I think vision api needs some vision.

**Comment by pm0733464** [comment]
**UPDATE:**
The Google Play Services 8.4 release lowers the amount of free storage required in order to enable the download.  This is now 500 MB or 10% of total storage, whichever is less.  See the release notes:

https://developers.google.com/vision/release-notes?hl=en

**Comment by xGary** [comment]
Glad there is progress on this issue.

I am curious, why not just allow the download as long as the user has enough space? So users don't run into low storage issues?

**Comment by pm0733464** [comment]
The reason is that we don't want users to get into a state where all of their storage is used, and basic functions of Android or other apps start to break due to a lack of storage.  For example, if there is no free storage, then the user wouldn't have enough storage to download app updates, Google Play Services updates, etc.

**Comment by gunhansancar** [comment]
Good news.
Thanks for the effort.

But, I second to @xGary about his thoughts anyway.

@pm0733464 Could you mind to share exactly how many bytes vision api require to download?

**Comment by pm0733464** [comment]
I agree that this still isn't the ideal solution, and we are discussing further improvements.

I don't have the exact numbers in front of me, but the barcode download is roughly 500KB and the face download is roughly 6MB.  It will only download what the apps on the device requires, rather than always downloading both of these.  And apps share these libraries, so each only needs to be downloaded once.

**Comment by jehy** [comment]
A year passed and still no updates on this issue? Can we at least get a friendly error from somewhere instead of manual checking with code from gist by @mitchross?

**Comment by capezzbr** [comment]
Three years passed 😢
